lib/puppet/application/agent.rb

```diff
@@ -71,6 +71,10 @@ def preinit
     options[:waitforcert] = arg.to_i
   end
 
+  option("--job-id ID") do |arg|
+    options[:job_id] = arg
```

Comment:
and no handlingdownstream ie in orchestrator will have any side-effects / security risks from taking in this string as-is?